src/modm/driver/encoder/as5600.hpp

```diff
@@ -0,0 +1,346 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2024, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <modm/architecture/interface/i2c_device.hpp>
+#include <modm/architecture/interface/register.hpp>
+
+#include "angle.hpp"
+
+namespace modm
+{
+
+struct as5600
+{
+
+	// @see datasheet page 19
+	enum Register : uint8_t
+	{
+		ZMCO = 0x00,
+		ZPOS = 0x01,       // two bytes
+		MPOS = 0x03,       // two bytes
+		MANG = 0x05,       // two bytes
+		CONF = 0x07,       // two bytes
+		ANGLE_RAW = 0x0C,  // two bytes
+		ANGLE = 0x0E,      // two bytes
+		STATUS = 0x0B,
+		AGC = 0x1A,
+		MAGNITUDE = 0x1B,  // two bytes
+		BURN = 0xFF,
+
+		// Only AS5600L supports changing the I2c address
+		I2C_ADDR = 0x20,
+		I2C_UPDT = 0x21,
+	};
+
+	// @see datasheet page 20
+	enum class Config : uint16_t
+	{
+		PM0 = Bit0,  // Power Mode
+		PM1 = Bit1,
+
+		HYST0 = Bit2,  // Hysteresis
+		HYST1 = Bit3,
+
+		OUTS0 = Bit4,  // Output Stage
+		OUTS1 = Bit5,
+
+		PWM0 = Bit6,  // PWM Frequency
+		PWM1 = Bit7,
+
+		SF0 = Bit8,  // Slow Filter
+		SF1 = Bit9,
+
+		FTH0 = Bit10,  // Fast Filter Threshold
+		FTH1 = Bit11,
+		FTH2 = Bit12,
+
+		WD = Bit13  // Watchdog    0: Off, 1: On
+	};
+	MODM_FLAGS16(Config);
+
+	enum class PowerMode : uint16_t
+	{
+		NOM = 0,
+		LPM1 = 1,
+		LPM2 = 2,
+		LPM3 = 3,
+	};
+	using PowerMode_t = Configuration<Config_t, PowerMode, 0b11, 0>;
+
+	enum class Hysteresis : uint16_t
+	{
+		OFF = 0,
+		LSB = 1,
+		LSB2 = 2,
+		LSB3 = 3
+	};
+	using Hysteresis_t = Configuration<Config_t, Hysteresis, 0b11, 2>;
+
+	enum class OutputStage : uint16_t
+	{
+		ANALOG = 0,
+		ANALOG_REDUCED = 1,
+		PWM = 2
+	};
+	using OutputStage_t = Configuration<Config_t, OutputStage, 0b11, 4>;
+
+	enum class PWMFrequency : uint16_t
+	{
+		_115Hz = 0,
```

Comment:
```suggestion
		Hz115 = 0,
```
In all other drivers we put the unit at the front. Not pretty, but better than the leading _